roadmap/implementers-guide/src/runtime/inclusion.md

```diff
@@ -68,13 +68,18 @@ All failed checks should lead to an unrecoverable error making the block invalid
   1. Transform each [`CommittedCandidateReceipt`](../types/candidate.md#committed-candidate-receipt) into the corresponding [`CandidateReceipt`](../types/candidate.md#candidate-receipt), setting the commitments aside.
   1. check the backing of the candidate using the signatures and the bitfields, comparing against the validators assigned to the groups, fetched with the `group_validators` lookup.
   1. check that the upward messages, when combined with the existing queue size, are not exceeding `config.max_upward_queue_count` and `config.watermark_upward_queue_size` parameters.
+  1. call `Router::ensure_processed_downward_messages(para, processed_downward_messages)` to check rules of processing the downward message queue.
```

Comment:
Other points in this section refer to "each candidate" because the parameter is a set of candidates. 

`processed_downwards_messages` is part of the `CandidateCommitments`? it would be more clear to reference that directly because the name is not defined above.

Reply:
> Other points in this section refer to "each candidate" because the parameter is a set of candidates.

ah good point, fixed.

> processed_downwards_messages is part of the CandidateCommitments? it would be more clear to reference that directly because the name is not defined above.

Ah I actually considered to do that but thought it was too verbose. I referred it as e.g. `commitments.processed_downward_messages`. Is that what you meant?

Reply:
Yeah that is reasonable